yorkie/src/main/kotlin/dev/yorkie/document/crdt/CrdtRoot.kt

```diff
@@ -128,12 +128,16 @@ internal class CrdtRoot(val rootObject: CrdtObject) {
      */
     fun garbageCollect(executedAt: TimeTicket): Int {
         var count = 0
-        removedElementSetByCreatedAt.forEach { createdAt ->
-            val pair = elementPairMapByCreatedAt[createdAt] ?: return@forEach
-            if (pair.element.isRemoved && pair.element.removedAt <= executedAt) {
-                pair.parent?.delete(pair.element)
-                count += garbageCollectInternal(pair.element)
+        try {
+            removedElementSetByCreatedAt.forEach { createdAt ->
+                val pair = elementPairMapByCreatedAt[createdAt] ?: return@forEach
+                if (pair.element.isRemoved && pair.element.removedAt <= executedAt) {
+                    pair.parent?.delete(pair.element)
+                    count += garbageCollectInternal(pair.element)
+                }
             }
+        } catch (e: Exception) {
+            // TODO(7hong): handle ConcurrentModificationException
```

Comment:
modifying a collection should not happen inside a loop.
we should use iterator.. all loop a snapshot of a collection.

Reply:
fixed this issue on #43